namerd/storage/consul/src/main/scala/io/buoyant/namerd/storage/consul/ConsulDtabStore.scala

```diff
@@ -62,7 +69,8 @@ class ConsulDtabStore(
     Activity(run).stabilize
   }
 
-  def create(ns: Ns, dtab: Dtab): Future[Unit] =
+  def create(ns: Ns, dtab: Dtab): Future[Unit] = {
+    if (!namespaceIsValid(ns)) Future.exception(new DtabNamespaceInvalidException(ns))
```

Comment:
I don't think this does anything.  I think you either need to `return Future.exception(...)` here or (my preference) put the rest of this method in an `else` block.